sync/rpc/server.go

```diff
@@ -807,22 +816,25 @@ func (s *SyncAgentServer) SnapshotPurge(ctx context.Context, req *SnapshotPurgeR
 				}
 				if info.Removed {
 					if info.Name == VolumeHeadName {
-						err = fmt.Errorf("BUG: Volume head was marked as removed")
+						s.PurgeStatus.Error = fmt.Errorf("BUG: Volume head was marked as removed")
 						return
 					}
 
-					if err = s.processRemoveSnapshot(snapshot); err != nil {
+					if s.PurgeStatus.Error = s.processRemoveSnapshot(snapshot, index, leafCount); s.PurgeStatus.Error != nil {
 						return
 					}
 				}
 				snapshot = info.Parent
 			}
 			// Update snapshotInfo in case some nodes have been removed
-			snapshotsInfo, err = s.getSnapshotsInfo()
-			if err != nil {
+			snapshotsInfo, s.PurgeStatus.Error = s.getSnapshotsInfo()
+			if s.PurgeStatus.Error != nil {
 				return
 			}
+			s.PurgeStatus.Progress = int((float32(index) + 1) / float32(leafCount) * 100)
```

Comment:
We only have one leaf normally... 

We can calculate the progress based on how many snapshots we need to remove.